Open Judge System/Services/OJS.Services/Business/ContestCategories/ContestCategoriesImportService.cs

```diff
@@ -0,0 +1,91 @@
+namespace OJS.Services.Business.ContestCategories
+{
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using OJS.Data.Models;
+    using OJS.Services.Business.Contests;
+    using OJS.Services.Business.Contests.Models;
+    using OJS.Services.Common.HttpRequester;
+
+    public class ContestCategoriesImportService : IContestCategoriesImportService
+    {
+        private readonly IContestsImportService contestsImportService;
+        private readonly IHttpRequesterService httpRequester;
+
+        public ContestCategoriesImportService(
+            IContestsImportService contestsImportService,
+            IHttpRequesterService httpRequester)
+        {
+            this.contestsImportService = contestsImportService;
+            this.httpRequester = httpRequester;
+        }
+
+        public async Task<string> ImportContestsIntoCategory(
+            int categoryId,
+            string ojsPlatformUrl,
+            bool replace,
+            string apiKey,
+            params int[] contestIds)
+        {
+            var contestImportResults = new List<ContestImportResult>();
```

Comment:
Depending on how much we want to optimize the performance and how large the data that is being processed is, here we can use a `Dictionary<bool, List<ContestImportResult>>`, instead of `List<ContestImportResult>` for `contestImportResults`. The boolean key would indicate whether the result was `successful` or `not`, this way we wouldn't have to filter the results at the end of the method:
```c#
var contestImportResults = new Dictionary<bool, List<ContestImportResult>>
{
    { true, new List<ContestImportResult>() },
    { false, new List<ContestImportResult>() }
};
    
    ................
    
var failedResults = contestImportResults[false];
var successfulResults = contestImportResults[true];
```
This can be done using two separate collections for the successful and unsuccessful results as well. Like mentioned, it depends on whether we have a large number of contests that is being imported:
```c#
var successfulResults = new List<ContestImportResult>();
var failedResults = new List<ContestImportResult>();
```

Reply:
The number of contests will not be large at all. The in memory filtration performance hit is insignificant compared to other factors like db access and http requests in this flow. The import is done in a background job anyway and speed is not crucial.